app/configurator/configurator-state/segment-config-state.tsx

```diff
@@ -0,0 +1,34 @@
+import get from "lodash/get";
+import { ChangeEvent } from "react";
+
+import { isSegmentInConfig } from "@/config-types";
+import { getChartConfig } from "@/config-utils";
+import { useConfiguratorState } from "@/src";
+import useEvent from "@/utils/use-event";
+
+import { isConfiguring } from ".";
+
+export const useLegendTitleVisibility = (target: "showTitle") => {
```

Comment:
Nit: I think we can remove `target` and hard-code the "showTitle" in the function body, I don't see the function being used in other contexts, at least at the moment :)